File: stack.py
class Stack:
    # Constructor
    def __init__(self):
        # set empty array where data will be pushed into
        self.data = []
        # this will keep track for how much elements are within our array
        self.length = 0

    def push(self, element):
        # append is just like push in javascript. Will append new element to the end of the array/list
        self.data.append(element)
        self.length += 1


    def Top(self):
        return self.data[len(self.data)-1]


    def Pop(self):
        if(self.length > 0):
            self.length -= 1
            return self.data.pop()


    def isEmpty(self):
        return self.length == 0


stack = Stack()

stack.push(1)
stack.push(5)
stack.push(3)
stack.push(4555)
print("length: ", stack.length)
stack.Pop()
print("length: ", stack.length)
print(stack.data)
print(stack.Top())

File: stack.js
class Stack {
  constructor() {
    this._data = [];
    this.length = 0;
  }

  push(element) {
    this._data.push(element);
    this.length++;
  }
  top() {
    return this._data[this._data.length - 1];
  }
  pop() {
    if (this.length > 0) {
      this.length--;
    }
    return this._data.splice(-1)[0];
  }
  isEmpty() {
      return this.length == 0;
  }
}

const stack = new Stack();
stack.push(1);
stack.push(14);
stack.push(122);
stack.push(20);
stack.push(20039);
console.log("length", stack.length); // length 5
console.log(stack.top()); // 20039
console.log(stack.pop()); // 20039 removed
console.log("length", stack.length); // length 4
console.log(stack.pop()); // 20 removed
console.log(stack.pop()); // 122 removed
console.log(stack.pop()); // 14 removed
console.log(stack.pop()); // 1 removed
console.log(stack.pop()); // undefined
console.log(stack.isEmpty()); // true